packages/recommend-react/src/useFrequentlyBoughtTogether.ts

```diff
@@ -17,32 +18,51 @@ export function useFrequentlyBoughtTogether<TObject>({
   recommendClient,
   threshold,
   transformItems: userTransformItems,
-}: GetFrequentlyBoughtTogetherProps<TObject>) {
-  const [result, setResult] = useState<GetRecommendationsResult<TObject>>({
-    recommendations: [],
-  });
+  initialState: userInitialState,
+}: GetFrequentlyBoughtTogetherProps<TObject> & {
+  initialState?: InitialState<TObject>;
+}) {
+  const isFirstRenderRef = useRef(true);
+
   const { status, setStatus } = useStatus('loading');
   const objectIDs = useStableValue(userObjectIDs);
   const transformItems = useStableValue(userTransformItems);
   const queryParameters = useStableValue(userQueryParameters);
 
+  const initialState = useStableValue<GetRecommendationsResult<TObject>>({
+    recommendations: [],
+    ...userInitialState,
+  });
+  const [result, setResult] = useState<GetRecommendationsResult<TObject>>(
+    initialState
+  );
+
   useAlgoliaAgent({ recommendClient });
 
   useEffect(() => {
-    setStatus('loading');
-    getFrequentlyBoughtTogether({
-      indexName,
-      maxRecommendations,
-      objectIDs,
-      queryParameters,
-      recommendClient,
-      threshold,
-      transformItems,
-    }).then((response) => {
-      setResult(response);
+    const shouldFetch = !userInitialState || !isFirstRenderRef.current;
+
+    if (shouldFetch) {
+      setStatus('loading');
+      getFrequentlyBoughtTogether({
+        indexName,
+        maxRecommendations,
+        objectIDs,
+        queryParameters,
+        recommendClient,
+        threshold,
+        transformItems,
+      }).then((response) => {
+        setResult(response);
+        setStatus('idle');
+      });
+    } else {
       setStatus('idle');
```

Comment:
We don't need to trigger a state change (and therefore a render) for this. We can directly initialize it in the state declaration:

```tsx
const { status, setStatus } = useStatus(userInitialState ? 'idle' : 'loading');
```

Reply:
keep forgetting to take re-renders into account, it's like it's my first day in React 🤦🏻‍♀️ 
update: https://github.com/algolia/recommend/pull/109/commits/bce36b23872ceeeb5ce80462e96ba983a9ff0760